File: 230117-dojo/java/app/src/main/java/Solution.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        // 1. 스택으로 풀어보기
        Stack<Integer> stack = new Stack<>();

        for (int number : arr) {
            if (stack.empty()) {
                stack.push(number);
                continue;
            }

            if (stack.peek() != number) {
                stack.push(number);
            }
        }

        int[] answer = new int[stack.size()];

        for (int i = answer.length - 1; i >= 0; i -= 1) {
            answer[i] = stack.pop();
        }

        return answer;
    }
}

File: 230117-dojo/javascript/solution.js
export default function solution(arr) {
  const answer = [];

  arr.forEach((value) => {
    if (answer.length === 0) {
      answer.push(value);
    }

    if (answer[answer.length - 1] !== value) {
      answer.push(value);
    }
  });

  return answer;
}

// export default function solution(arr) {
//     return arr.filter((value, index) => value !== arr[index + 1]);
//   }
